src/AutoPilotPlugins/Common/MotorComponent.qml

```diff
@@ -222,13 +222,12 @@ SetupPage {
                 spacing: ScreenTools.defaultFontPixelWidth
                 Switch {
                     id: safetySwitch
-                    onClicked: {
+                    onToggled: {
                         if (controller.vehicle.armed) {
                             timer.stop()
                         }
 
                         controller.vehicle.armed = checked
-                        checked = controller.vehicle.armed // As crazy as this looks, it keeps things working the way they should see onArmedChanged below, that will take care of checked state
```

Comment:
I overlooked this section of the patch before merging. Did you test this?

I don't recall exactly why I did it this way, but what if the vehicle denies the arming command (I think this might have been it)?

Reply:
I just tested it,  I removed it because it causes a weird behaviour when dragging:

![deepin-screen-recorder_select area_20180920205005](https://user-images.githubusercontent.com/4013804/45852564-c8c3b780-bd16-11e8-918f-ded9c1dea470.gif)
But you are right, this way the toggle can stay on even if the vehicle refuses to arm. I'll take another look into it.